File: Exercise 4/task2/hello.cpp
#include <string.h>
#include <iostream>
#include <stdio.h>
#include <string>
using namespace std;

string update_string(string input) {
    
    char input_array[input.length() + 1];
    
    strncpy(input_array, input.c_str(), sizeof(input_array));
    
    string return_string = "";
    
    for (char c : input_array)
    {
        if (c == '&')
        {
            return_string += "&amp;";
        }
        else if (c == '<')
        {
            return_string += "&lt;";
        }
        else if (c == '>')
        {
            return_string += "&gt;";
        }
        else
        {
            return_string += c;
        }
    }
    return return_string;
}

int main(int argc, char const *argv[])
{
    cout << "Enter input:\n";
    string input;
    cin >> input;
    cout << "Result:\n"
         << update_string(input) + "\n";
    return 0;
}

File: Exercise 4/task2/hello.rs
fn update_string(input: String) -> String{
    let mut return_string = String::new();
    let input_vec: Vec<char>  = input.chars().collect();
    for c in input_vec {
        match c{
            '&' => return_string+="&amp;",
            '>' => return_string+="&gt;",
            '<' => return_string+="&lt;",
            _ => return_string.push(c)
        }
    }
    return return_string;
}

fn main(){
    let mut line = String::new();
    println!("Enter input:");
    std::io::stdin().read_line(&mut line).unwrap();
    let result = update_string(line);
    println!("Result: \n{}",result);
}   
